### PR TITLE
Fix bug #37782

### LINKED ISSUE #37782: BUG: json_normalize cannot parse metadata fields list type
#### Code Sample, a copy-pastable example

```python
test_data = [
    {"values": [1, 2, 3], "metadata": {"listdata": [1, 2]}}]

df = json_normalize(test_data,
                    record_path=["values"],
                    meta=[["metadata", "listdata"]])
print(df)

```

#### Problem description

It throws error ```ValueError: Length of values (6) does not match length of index (3)```

Changing listdata field from `[1,2]` to `{1,2}` helped but I can't control JSON I receive.

#### Expected Output

| 0 | metadata.listdata |
|---|-------------------|
| 1 | [1, 2]            |
| 2 | [1, 2]            |
| 3 | [1, 2]            |

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.6.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-52-generic
Version          : #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.1.4
numpy            : 1.19.4
pytz             : 2020.4
dateutil         : 2.8.1

</details>

### PR DESCRIPTION
**Pull request by felipemaion**
- [X] closes #37782 
- [X] [Tests added and passed]
- [X] All [code checks passed]
- [X] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file.

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -372,10 +372,10 @@ I/O
 ^^^
 - :meth:`DataFrame.to_orc` now raising ``ValueError`` when non-default :class:`Index` is given (:issue:`51828`)
 - :meth:`DataFrame.to_sql` now raising ``ValueError`` when the name param is left empty while using SQLAlchemy to connect (:issue:`52675`)
+- Bug in :func:`json_normalize`, json_normalize cannot parse metadata fields list type (:issue:`#37782`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:375]:
```suggestion
- Bug in :func:`json_normalize`, json_normalize cannot parse metadata fields list type (:issue:`37782`)
```

File: pandas/tests/io/json/test_normalize.py [hunk 2]
```diff
@@ -137,6 +137,11 @@ def max_level_test_input_data():
     ]
 
 
+@pytest.fixture
+def parse_metadata_fields_list_type():
```
Comment (mroeschke) [inline pandas/tests/io/json/test_normalize.py:141]: Can you just hardcode this in the function?

File: pandas/io/json/_normalize.py [hunk 3]
```diff
@@ -535,5 +535,15 @@ def _recursive_extract(data, path, seen_meta, level: int = 0) -> None:
             raise ValueError(
                 f"Conflicting metadata name {k}, need distinguishing prefix "
             )
-        result[k] = np.array(v, dtype=object).repeat(lengths)
+        #### FIX BUG #37782: https://github.com/pandas-dev/pandas/issues/37782
```
Comment (mroeschke) [inline pandas/io/json/_normalize.py:538]:
```suggestion
        #GH 37782
```

File: pandas/tests/io/json/test_normalize.py (hunk 2 above)
Comment (felipemaion) [reply pandas/tests/io/json/test_normalize.py:141]: @mroeschke like that?
`    def test_fields_list_type_normalize(self):
        parse_metadata_fields_list_type=[{"values": [1, 2, 3], "metadata": {"listdata": [1, 2]}}]
        result = json_normalize(
            parse_metadata_fields_list_type,
            record_path=["values"],
            meta=[["metadata", "listdata"]],
        )
        expected = DataFrame(
            {0: [1, 2, 3], "metadata.listdata": [[1, 2], [1, 2], [1, 2]]}
        )
        tm.assert_frame_equal(result, expected)`
        
        Should I push it? I did this as function just by looking other examples in the file.
        Let me know.

Comment (mroeschke) [reply pandas/tests/io/json/test_normalize.py:141]: Yes

Comment (felipemaion) [reply pandas/tests/io/json/test_normalize.py]: @mroeschke I just did it. But things looks different. I'm not sure if it was pushed properly. Can you check please? 
(My first PR for a big project)

File: pandas/io/json/_normalize.py (hunk 3 above)
Comment (mroeschke) [reply pandas/io/json/_normalize.py:538]: For this one, you can click `Commit suggestion`

File: doc/source/whatsnew/v2.1.0.rst (hunk 1 above)
Comment (felipemaion) [reply doc/source/whatsnew/v2.1.0.rst]: @mroeschke I'm seeing the message: This branch has conflicts that must be resolved
Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.
Conflicting files
doc/source/whatsnew/v2.1.0.rst

Can you please take a look and guide me on this?

Comment (mroeschke) [reply doc/source/whatsnew/v2.1.0.rst]: Please follow these constructions to update your branch and fix the conflict: https://pandas.pydata.org/docs/development/contributing.html#updating-your-pull-request

Comment (felipemaion) [reply doc/source/whatsnew/v2.1.0.rst]: @mroeschke is there anything else that I have to do? I only see a "Close pull request" button at bottom of the page. This should be pressed?? (Sorry, just afraid to mess anything else!)

File: doc/source/whatsnew/v2.1.0.rst [hunk 4]
```diff
@@ -374,11 +374,13 @@ I/O
 ^^^
 - :meth:`DataFrame.to_orc` now raising ``ValueError`` when non-default :class:`Index` is given (:issue:`51828`)
 - :meth:`DataFrame.to_sql` now raising ``ValueError`` when the name param is left empty while using SQLAlchemy to connect (:issue:`52675`)
+- Bug in :func:`json_normalize`, fix json_normalize cannot parse metadata fields list type (:issue:`37782`)
 - Bug in :func:`read_hdf` not properly closing store after a ``IndexError`` is raised (:issue:`52781`)
 - Bug in :func:`read_html`, style elements were read into DataFrames (:issue:`52197`)
 - Bug in :func:`read_html`, tail texts were removed together with elements containing ``display:none`` style (:issue:`51629`)
 - Bug in displaying a :class:`MultiIndex` with a long element (:issue:`52960`)
 
+
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:383]:
```suggestion
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @felipemaion

**Pull request merged**